modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/ZonePartitionReplicaListener.java

```diff
@@ -138,26 +148,23 @@ public ZonePartitionReplicaListener(
                 replicationGroupId
         );
 
-        minimumActiveTxTimeReplicaRequestHandler = new MinimumActiveTxTimeReplicaRequestHandler(
-                clockService,
-                raftCommandApplicator
-        );
-
-        vacuumTxStateReplicaRequestHandler = new VacuumTxStateReplicaRequestHandler(raftCommandApplicator);
-
         txStateCommitPartitionReplicaRequestHandler = new TxStateCommitPartitionReplicaRequestHandler(
                 txStatePartitionStorage,
                 txManager,
                 clusterNodeResolver,
                 localNode,
-                new TxRecoveryEngine(
-                        txManager,
-                        clusterNodeResolver,
-                        replicationGroupId,
-                        ZonePartitionReplicaListener::createAbandonedTxRecoveryEnlistment
-                )
+                txRecoveryEngine
         );
 
+        txRecoveryMessageHandler = new TxRecoveryMessageHandler(txStatePartitionStorage, replicationGroupId, txRecoveryEngine);
+
+        minimumActiveTxTimeReplicaRequestHandler = new MinimumActiveTxTimeReplicaRequestHandler(
+                clockService,
+                raftCommandApplicator
+        );
+
+        vacuumTxStateReplicaRequestHandler = new VacuumTxStateReplicaRequestHandler(raftCommandApplicator);
```

Comment:
Can I ask you to delete an excessive newline whitespace inside `VacuumTxStateReplicaRequestHandler#handle`?